backend/src/index.spec.ts

```diff
@@ -58,105 +75,155 @@ describe('Test apollo server queries', () => {
       { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
     ]
     const userData = [
-      { name: 'user1', posts: [postData[0]] }
+      { id: 'userid1', name: 'user1', email: 'user1@example.org', password: bcrypt.hashSync('user1password', 10), posts: [postData[0]] }
     ]
     postData[0].author = userData[0]
 
-    const server = new ApolloServer({
-      typeDefs,
-      resolvers,
-      dataSources: () => ({
-        posts: new PostsDataSource(postData, userData)
-      })
-    })
+    const { query } = setupServer(postData, userData)
     const GET_USERS = '{ users { name }}'
-    const { query } = createTestClient(server)
     const res = await query({ query: GET_USERS })
     expect(res.errors).toBeUndefined()
     expect(res.data.users.length).toEqual(1)
     expect(res.data.users[0].name).toEqual('user1')
   })
 
+  it('upvote a post with invalid token returns error', async () => {
+    const postData = [
+      { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
```

Comment:
I would initialize `voters` and `author` in the constructor of a `Post` class. You could re-use that class in the test:
```suggestion
      new Post({ id: 'post1', title: 'Item 1' })
```